qemu/tests/cfg/hotplug_virtio_mem.cfg

```diff
@@ -29,3 +29,4 @@
     pcie_extra_root_port = 1
     requested-size_test_vmem0 = "4G 2G"
     error_msg = "virtio-mem device cannot get unplugged while 'size' != '0'"
+    new_error_msg = "virtio-mem device cannot get unplugged while some of its memory is still plugged"
```

Comment:
We could still use one parameter `error_msg` to represent different version's error messages by the regex expression: `re.search`.
For instance: 
`error_msg = "virtio-mem device cannot get unplugged while ('size' != '0')|(some of its memory is still plugged)"`

Hi @mcasquer 
How about it? Please let me know your thoughts.
Thanks.

Reply:
@YongxueHong Updated, thanks !

Reply:
Hi @mcasquer and @YongxueHong , I might have some concerns about this part, think about this situation: 

qemu 9.1, the error msg somehow doesn't change to the latest one, the test case will still pass, which is not expected, it should fail.

So what about using RHEL version filters(we don't have qemu version filters) to set different error messages for different qemu versions, 
```
error_msg = "virtio-mem device cannot get unplugged while some of its memory is still plugged"
Host_RHEL.m<xx>.u<yy>: >>> RHEL versions supporting the old error msg
    error_msg = "virtio-mem device cannot get unplugged while 'size' != '0'"
```

What do you think?

Reply:
> Hi @mcasquer and @YongxueHong , I might have some concerns about this part, think about this situation:
> 
> qemu 9.1, the error msg somehow doesn't change to the latest one, the test case will still pass, which is not expected, it should fail.

Hi @zhencliu 
Could you explain in more detail? I did not get your point.
As my understanding, The error meg should be `virtio-mem device cannot get unplugged while some of its memory is still plugged` with qemu 9.1 and than we could check if it match the expected regex expression `error_msg = "virtio-mem device cannot get unplugged while ('size' != '0')|(some of its memory is still plugged)"` by re.search.

> 
> So what about using RHEL version filters(we don't have qemu version filters) to set different error messages for different qemu versions,
> 
> ```
> error_msg = "virtio-mem device cannot get unplugged while some of its memory is still plugged"
> Host_RHEL.m<xx>.u<yy>: >>> RHEL versions supporting the old error msg
>     error_msg = "virtio-mem device cannot get unplugged while 'size' != '0'"
> ```
> 
> What do you think?

Reply:
> Hi @mcasquer and @YongxueHong , I might have some concerns about this part, think about this situation:
> 
> qemu 9.1, the error msg somehow doesn't change to the latest one, the test case will still pass, which is not expected, it should fail.
> 
> So what about using RHEL version filters(we don't have qemu version filters) to set different error messages for different qemu versions,
> 
> ```
> error_msg = "virtio-mem device cannot get unplugged while some of its memory is still plugged"
> Host_RHEL.m<xx>.u<yy>: >>> RHEL versions supporting the old error msg
>     error_msg = "virtio-mem device cannot get unplugged while 'size' != '0'"
> ```
> 
> What do you think?

@zhencliu we can do the QEMU version filter in the python script (as I implemented at first), but filtering by RHEL version could be a problem rather than a solution. IMHO the @YongxueHong solution is the best.

Reply:
Sure, I am OK with then.